language/movie_nl/main.py

```diff
@@ -0,0 +1,366 @@
+        return 1
+    else:
+        return 0
+
+
+def process_movie_reviews(service, reader, sentiment_writer, entity_writer):
+    """Perform some sentiment math and come up with movie review."""
+    collected_entities = {}
+
+    for document in reader:
+        try:
+            sentiment_total, entities = get_sentiment_entities(
+                service, document)
+        except HttpError as e:
+            logging.error("Error in process_movie_reviews {}".format(e.content))
+            continue
+
+        document.label = get_sentiment_label(sentiment_total)
+
+        sentiment_writer.write(
+            to_sentiment_json(
+                document.doc_id,
+                sentiment_total,
+                document.label
+            )
+        )
+
+        sentiment_writer.write('\n')
+
+        for ent in entities:
+            ent_sent, frequency = collected_entities.get(ent, (0, 0))
+            ent_sent += sentiment_total
+            frequency += 1
+
+            collected_entities[ent] = (ent_sent, frequency)
+
+    for entity, e_tuple in collected_entities.items():
+        entity_writer.write(to_entity_json(entity, e_tuple))
+        entity_writer.write('\n')
+
+    sentiment_writer.flush()
+    entity_writer.flush()
+
+
+def document_generator(dir_path_pattern, count=None):
+    """Generator for the input movie documents."""
+    for running_count, item in enumerate(glob.iglob(dir_path_pattern)):
+        if count and running_count >= count:
+            raise StopIteration()
+
+        doc_id = os.path.basename(item)
+
+        with codecs.open(item, encoding='utf-8') as f:
+            try:
+                text = f.read()
+            except UnicodeDecodeError:
+                continue
+
+            yield Document(text, doc_id, item)
+
+
+def rank_entities(reader, sentiment=None, topn=None, reverse_bool=False):
+    """Rank the entities (actors) based on their sentiment
+    assigned from the movie."""
+
+    items = []
+    for item in reader:
+        json_item = json.loads(item)
+        sent = json_item.get('sentiment')
+        entity_item = (sent, json_item)
+
+        if sentiment:
+            if sentiment == 'pos' and sent > 0:
+                items.append(entity_item)
+            elif sentiment == 'neg' and sent < 0:
+                items.append(entity_item)
+        else:
+            items.append(entity_item)
+
+    items.sort(reverse=reverse_bool)
+    items = [json.dumps(item[1]) for item in items]
+
+    if topn:
+        print('\n'.join(items[:topn]))
+    else:
+        print('\n'.join(items))
+
+
+def get_service():
+    """Build a client to the Google Cloud Natural Language API."""
+
+    credentials = GoogleCredentials.get_application_default()
+    scoped_credentials = credentials.create_scoped(
+        ['https://www.googleapis.com/auth/cloud-platform'])
+    http = httplib2.Http()
+    scoped_credentials.authorize(http)
+    return discovery.build('language', 'v1beta1', http=http)
```

Comment:
FYI now that the api is public, you can just do:

    return discovery.build('language', 'v1beta1' ,credentials=GoogleCredentials.get_application_default())